test/common/signal/fatal_action_test.cc

```diff
@@ -23,17 +26,22 @@ class TestFatalErrorHandler : public FatalErrorHandlerInterface {
   void onFatalError(std::ostream& /*os*/) const override {}
   void
   runFatalActionsOnTrackedObject(const FatalAction::FatalActionPtrList& actions) const override {
-    // Call the Fatal Actions with nullptr
+    // Call the Fatal Actions with a non-empty vector so it runs the action.
+    std::vector<const ScopeTrackedObject*> tracked_objects{nullptr};
     for (const Server::Configuration::FatalActionPtr& action : actions) {
-      action->run(nullptr);
+      action->run(tracked_objects);
     }
   }
 };
 
 class TestFatalAction : public Server::Configuration::FatalAction {
 public:
   TestFatalAction(bool is_safe, int* const counter) : is_safe_(is_safe), counter_(counter) {}
-  void run(const ScopeTrackedObject* /*current_object*/) override { ++(*counter_); }
+  void run(const std::vector<const ScopeTrackedObject*>& tracked_objects) override {
+    for (size_t i = 0; i < tracked_objects.size(); ++i) {
+      ++(*counter_);
```

Comment:
Seems like you have no test cases with tracked_objects.size() != 1

Reply:
Good point, removed. Doesn't make sense to me to try to add other test cases with >1 tracked object size here as it'd be testing whether the test fatal action uses it.